### PR TITLE
Optimization: asynchronous writeback

### PR DESCRIPTION
**Pull request by sourcejedi**
fsync() after each segment write is suboptimal :).  It means you stop (cpu) processing to wait for the physical disk write. And the default segment size is 5MB. (I noticed bup avoids this issue by writing pack files of 1GB by default :).
    
Improvements will vary depending disk/cpu speed.  (I guess the worst case was when they were evenly matched).
    
 * Writing 65M on SheevaPlug "NAS" went from 47s to 45s.
 * 920M on desktop HDD (read from SSD) went from 68s to 45s

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
Not sure whether I should / would accept this code.
While it's doing the right thing, it is kind of a fraction what the (still experimental) code in multithreading branch does (see my repo). So maybe rather that code should be refined further / tested more to solve a lot of scheduling problems, not just one.

**Comment by sourcejedi** [comment]
I see your point :).  This is clearly incremental.  E.g. I don't know whether the code would get refactored when expanding the use of threading, so all the threading code can use the same primitives and style.

I do think the concept is right.  E.g. you should still do this if you were writing multiple segment files in parallel, you shouldn't just assume their blocking IO will be fortuitously out-of-sync.

**Comment by sourcejedi** [comment]
Sorry, I meant to also say I will look at your threading code.

**Comment by ThomasWaldmann** [comment]
for master branch (not multithreading), could we get a similar speedup with less code change by going from 5MB to e.g. 50MB?

File: borg/repository.py [hunk 1]
```diff
@@ -622,16 +626,112 @@ def write_commit(self):
...
+            raise e
+
+    def close(self):
+        try:
+            self.flush()
+        finally:
+            self.channel.put(None)  # tell thread to shutdown
+
+
+class Channel(object):
```
Comment (ThomasWaldmann) [inline borg/repository.py]: hmm, couldn't we just use a normal Queue with length 1 and just signal "task done" when we really have done it?

**Comment by ThomasWaldmann** [comment]
Just as a side note:

I am currently doing a large-scale backup for testing. The fsync somehow slows it down unnecessarily.

So to speed it up, I just removed the fsync in my local version, it's now 2x faster.

**Comment by ThomasWaldmann** [comment]
I am rejecting this for now and for master.

Multithreading code should go into multithreading branch (if not already implemented there) and will need a lot of testing before it gets merged into master.

**Pull request closed without merging**